push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/ApnsUtil.java

```diff
@@ -79,12 +86,22 @@ public static boolean checkValidity(final byte[] keystore, final char[] password
             while (aliases.hasMoreElements()) {
                 final String alias = aliases.nextElement();
                 final X509Certificate certificate = (X509Certificate) keyStore.getCertificate(alias);
+                final X500Principal subjectX500Principal = certificate.getSubjectX500Principal();
+                final String subject = subjectX500Principal.getName();
 
-                try{
-                    certificate.checkValidity();
-                } catch (CertificateExpiredException | CertificateNotYetValidException e) {
-                    LOGGER.error("Provided APNs .p12 file is expired or not yet valid");
-                    return false;
+                final Matcher matcher = COMMON_NAME_PATTERN.matcher(subject);
```

Comment:
I believe we have bouncycastle in the list of UPS dependencies. If that is the case, a more robust way to extract the CN would be:
```java
X500Name x500name = new JcaX509CertificateHolder(cert).getSubject();
RDN cn = x500name.getRDNs(BCStyle.CN)[0];
```

Reply:
@ziccardi thanks, i will try this